props/saber_BC_buttons.h

```diff
@@ -383,34 +1158,253 @@ EFFECT(voldown);    // for decrease volume
 EFFECT(volmin);     // for minimum volume reached
 EFFECT(volmax);     // for maximum volume reached
 #ifdef ENABLE_FASTON
-#warning The faston.wav sound will be replaced with fastout.wav. If you have a good reason to keep faston.wav as is, please post at https://crucible.hubbe.net/
+// #warning The faston.wav sound will be replaced with fastout.wav. If you have a good reason to keep faston.wav as is, please post at https://crucible.hubbe.net/
 EFFECT(faston);     // for EFFECT_FAST_ON. Being replaced by fastout.wav, which is already defined in the main OS.
 #endif
-EFFECT(blstbgn);    // for Begin Multi-Blast
-EFFECT(blstend);    // for End Multi-Blast
 EFFECT(push);       // for Force Push gesture
-EFFECT(monosfx);    // for Monophonically played sounds (iceblade, seismic charge etc...)
-EFFECT(swap);       // for standalone triggering EffectSequence<>
+EFFECT(tr);         // for EFFECT_TRANSITION_SOUND, use with User Effects.
+EFFECT2(trloop, trloop);  // have YET to play with this
 EFFECT(mute);       // Notification before muted ignition to avoid confusion.
+EFFECT(mzoom);      // for Spam Blast enter/exit
+
+#ifndef MENU_SPEC_TEMPLATE
+template<class SPEC>
+struct BCVolumeMode : public SPEC::SteppedMode {
+  const int max_volume_ = VOLUME;
+  const int min_volume_ = VOLUME * 0.10;
+  float initial_volume_ = 0.0;
+  int initial_percentage_ = 0;
+  int percentage_ = 0;
+  bool announced_ = false;
+
+  int steps_per_revolution() override {
+    return 12;  // adjust for sensitivity
+  }
 
-class DelayTimer {
-public:
-    DelayTimer() : triggered_(false), trigger_time_(0), duration_(0) {}
+  void mode_activate(bool onreturn) override {
+    PVLOG_NORMAL << "** Enter Volume Menu\n";
+    initial_volume_ = dynamic_mixer.get_volume();
+    initial_percentage_ = round((initial_volume_ / max_volume_) * 10) * 10;
+   SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    mode::getSL<SPEC>()->SayEnterVolumeMenu();
+    SPEC::SteppedMode::mode_activate(onreturn);
+  }
+
+  void mode_deactivate() override {
+    if (percentage_ <= 10) {
+      mode::getSL<SPEC>()->SayMinimumVolume();
+    } else if (percentage_ >=100) {
+      mode::getSL<SPEC>()->SayMaximumVolume();
+    } else {
+      mode::getSL<SPEC>()->SayWhole(percentage_);
+      mode::getSL<SPEC>()->SayPercent();
+    }
+    mode::getSL<SPEC>()->SayVolumeMenuEnd();
+    SPEC::SteppedMode::mode_deactivate();
+  }
 
-    void trigger(uint32_t duration) {
-        triggered_ = true;
-        trigger_time_ = millis();
-        duration_ = duration;
+  void next() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ < max_volume_) {
+      current_volume_ += max_volume_ * 0.10;
```

Comment:
current_volume_ can end up higher than max_volume_ here.
Maybe add:
if (current_volume_ > max_volume_) current_volume_ = max_volume_;

(Or some std::min call)

Reply:
Perfect, thanks. Added this here, and also limited the min in `prev()`